hack/tools/release/notes.go

```diff
@@ -73,6 +73,8 @@ var (
 
 	prefixAreaLabel = flag.Bool("prefix-area-label", true, "If enabled, will prefix the area label.")
 
+	releaseCandidate            = flag.Bool("release-candidate", false, "If enabled, will add a release candidate warning header. (default false)")
+	deprecation                 = flag.Bool("deprecation", false, "If enabled, will add a templated deprecation warning header. (default false)")
```

Comment:
I would prefer if this was true by default] and when not needed, we can just remove the section from the notes.
The idea behind these changes is to prevent us from forgetting something in the notes. If the section is there and we don't have content, we won't miss deleting it. However, it's possible that we might miss passing this flag.

Reply:
That's fair - do you think there's even value to have the flag then vs just always templating this out?